app/views/smart_proxies/plugins/_puppet.html.erb

```diff
@@ -0,0 +1,29 @@
+<% dashboard = Dashboard::Data.new("puppetmaster = #{@smart_proxy.name}") %>
+<% @hosts = dashboard.hosts %>
+<% @report = dashboard.report %>
+<h3><%= _("Puppet") %></h3>
+<%= show_feature_version('puppet') %>
+<div class="col-md-2"><%= _('Hosts managed:') %></div>
+<div class="col-md-10">
+  <%= link_to @hosts.length, hosts_path(:search => "puppetmaster = \"#{@smart_proxy.name}\"") %>
+</div>
+<ul id="proxy-puppet-tab" class="nav nav-tabs nav-tabs-pf">
```

Comment:
Double tab navigation (we have tabs for features + tabs for reports & statistics) is usually known as an UI antipattern. How about showing all sections, with 3 headers? General, Environments, Activity